docs/deployment/incremental_collection.md

```diff
@@ -0,0 +1,129 @@
+<!--
+ - Licensed to the Apache Software Foundation (ASF) under one or more
+ - contributor license agreements.  See the NOTICE file distributed with
+ - this work for additional information regarding copyright ownership.
+ - The ASF licenses this file to You under the Apache License, Version 2.0
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+
+<div align=center>
+
+![](../imgs/kyuubi_logo.png)
+
+</div>
+
+# Solution for big result set
+
+Normally, when user sumbits a SELECT query to Spark SQL engine, the Driver calls `collect` to trigger calculation and
+retrieve all partitions from all Worker nodes, after all partitions data arrived, then Driver sends the data back to
+client through Kyuubi Server streamingly in small batch, the batch size decided by `TFetchResultsReq.maxRows`.
+
+Therefore, for query has big result set, the bottleneck is the Spark Driver, to avoid OOM, Spark has a configuration
```

Comment:
Therefore, for query has big result set, the bottleneck is the Spark Driver, -> Therefore, the bottleneck is the Spark Driver for a query with a big result set.

Reply:
to avoid OOM -> To avoid OOM